### PR TITLE
Slight improvements to Registry

### PR DESCRIPTION
**Pull request by ptare**
In particular:
* Remove unused enum value `wasValid`
* Use Natspec for documentation
* Add Registry to migrations
* Add an extra boolean to double enforce that roles are initialized only
once

Note that the migration will have to move much later in the sequence,
because we'd probably want to authorize the TokenizedDerivativeCreator.

Fixes #409

### CODE REVIEW

File: v1/contracts/Registry.sol [hunk 1]
```diff
@@ -8,6 +8,10 @@ import "openzeppelin-solidity/contracts/math/SafeMath.sol";
 pragma experimental ABIEncoderV2;
 
 
+/**
+ * @title Registry for derivatives and approved derivative creators
```
Comment (mrice32) [inline v1/contracts/Registry.sol]: nit: add `.`.

File: v1/migrations/4_deploy_registry.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+const Registry = artifacts.require("Registry");
+const { getKeysForNetwork, deployAndGet, addToTdr } = require("../../common/MigrationUtils.js");
+
+module.exports = async function(deployer, network, accounts) {
```
Comment (mrice32) [inline v1/migrations/4_deploy_registry.js]: While you're adding the deployment, could you copy over the deployment verification for the v0 registry from https://github.com/UMAprotocol/protocol/blob/2fb468d7c8391e9ef9893bdb9311012deb0a3308/v0/scripts/CheckDeploymentValidity.js#L19-L21 to https://github.com/UMAprotocol/protocol/blob/master/v1/scripts/CheckDeploymentValidity.js.

File: v1/contracts/Registry.sol (hunk 1 above)
Comment (ptare) [reply v1/contracts/Registry.sol]: Done.

File: v1/migrations/4_deploy_registry.js (hunk 2 above)
Comment (ptare) [reply v1/migrations/4_deploy_registry.js]: Ran:
`$(npm bin)/truffle migrate --reset --network=test`
and then:
` $(npm bin)/truffle exec scripts/CheckDeploymentValidity.js --network=test
Using network 'test'.

Deployment looks good!
`

**Pull request merged**